auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java

```diff
@@ -347,6 +350,33 @@ public boolean checkRight(String actorCrn, String userCrn, String right, Optiona
         return checkRight(actorCrn, userCrn, right, null, requestId);
     }
 
+    /**
+     * Retrieves whether the member has the specified rights.
+     *
+     * @param actorCrn the CRN of the actor
+     * @param memberCrn the CRN of the member
+     * @param rightChecks the rights to check
+     * @param requestId an optional request id
+     * @return a list of booleans indicating whether the member has the specified rights
+     */
+    public List<Boolean> hasRights(String actorCrn, String memberCrn, List<AuthorizationProto.RightCheck> rightChecks, Optional<String> requestId) {
+        LOGGER.info("Checking whether member [] has rights [{}]",
+                memberCrn,
+                rightChecks.stream().map(AuthorizationProto.RightCheck::getRight).collect(Collectors.toList()));
+        if (InternalCrnBuilder.isInternalCrn(memberCrn)) {
+            LOGGER.info("InternalCrn has all rights");
+            Boolean[] values = new Boolean[rightChecks.size()];
+            Arrays.fill(values, Boolean.TRUE);
+            return Arrays.asList(values);
+        }
+        try (ManagedChannelWrapper channelWrapper = makeWrapper()) {
+            AuthorizationClient client = new AuthorizationClient(channelWrapper.getChannel(), actorCrn);
+            List<Boolean> retVal = client.hasRights(requestId.orElse(UUID.randomUUID().toString()), memberCrn, rightChecks);
+            LOGGER.info("member {} has rights {}", memberCrn, retVal);
+            return retVal;
+        }
+    }
+
```

Comment:
So I prefer not to do the right check here, in the grpc ums client. Let's not optimize this right now. I think that the right way to optimize this is to have distributed policies that have ttl on them. The policy for an internal actor can be "forever" so that the clients will never go to the authorization service. But I do not want to copy the authorization checks all over the place.